android/app/src/main/cpp/CMakeLists.txt

```
@@ -0,0 +1,57 @@
+# For more information about using CMake with Android Studio, read the
+# documentation: https://d.android.com/studio/projects/add-native-code.html
+
+# Sets the minimum version of CMake required to build the native library.
+
+cmake_minimum_required(VERSION 3.18.1)
+
+# Declares and names the project.
+
+project("netguard")
+
+# Creates and names a library, sets it as either STATIC
+# or SHARED, and provides the relative paths to its source code.
+# You can define multiple libraries, and CMake builds them for you.
+# Gradle automatically packages shared libraries with your APK.
+
+add_library( netguard
+             SHARED
+             ../../../../../src/netguard.c
+             ../../../../../src/session.c
+             ../../../../../src/ip.c
+             ../../../../../src/tls.c
+             ../../../../../src/tcp.c
+             ../../../../../src/udp.c
+             ../../../../../src/icmp.c
+             ../../../../../src/dns.c
+             ../../../../../src/dhcp.c
+             ../../../../../src/pcap.c
+             ../../../../../src/memory.c
+             ../../../../../src/socks5.c
+             ../../../../../src/util.c
```

Quick note to sync with the latest netguard change:

```suggestion
             ../../../../../src/uid_mapping.c
             ../../../../../src/util.c
```

---

Done